### ISSUE TITLE
Hi, I tried to destroy and create a new Session with these lines in a coroutine :

### ISSUE THREAD

**Issue by abudriaz** [google-ar/arcore-unity-sdk#446]
```
ARCoreSession session = goARCoreDevice.GetComponent<ARCoreSession>();
        ARCoreSessionConfig myConfig = session.SessionConfig;
        Destroy(session);

        yield return null;

        session = goARCoreDevice.AddComponent<ARCoreSession>();
        session.SessionConfig = myConfig;
        session.enabled = true;
```

However, when I do this, the app is frozen for several seconds. Then, sometimes it works as expected, but sometimes the camera doesn't work anymore and is frozen on the last frame captured before the creation of a new session. 

Is there another way to reset the scan from the beginning (and not keep the previous scanning data) ?
Tested with ARCore SDK for Unity v1.2.0 on Galaxy S7 and S7 Edge.

_Originally posted by @NicolasWipon in https://github.com/google-ar/arcore-unity-sdk/issues/181#issuecomment-398322188_

**Comment by abudriaz** [comment]
you added these lines in which class?

**Comment by JanneMK** [comment]
Use DestroyImmediate(session);

**Comment by NicolasWipon** [comment]
Hi,
I added it in a personal script, it doesn't matter. I just called it when I wanted to reset the AR session. For information, "goARCoreDevice" is the root gameobject of the ARCoreDevice prefab instanciated in the scene, defined in the inspector of the MonoBehaviour with this line "public GameObject goARCoreDevice".

**Comment by AbdullahBurhan** [comment]
I think the freezing problem was solved in ARCore 1.5 version.
Try to use the 1.5 version, that could solve the problem.

**Comment by NicolasWipon** [comment]
Hi,
Yes, it seems to work with 1.5 version. :-)
I couldn't reproduce the problem for the moment.
